src/components/question_creation/question-types/coding-question.tsx

```
@@ -0,0 +1,582 @@
+  onExplanationChange: (explanation: string) => void;
+  onShowExplanationChange: (showExplanation: boolean) => void;
+  onFunctionMetadataChange?: (metadata: FunctionMetadata) => void;
+}
+
+// Utility function to generate boilerplate code
+const generateBoilerplateCode = (metadata: FunctionMetadata): string => {
+  const { name, parameters, returnType, language } = metadata;
+
+  switch (language) {
+    case "python":
+      const pythonParams = parameters.map((p) => p.name).join(", ");
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)